### DISCUSSION TITLE
Asset path specification? What does it all mean? :)

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#6884] [Q&A]
Hi,

On the following page concerning assets in Avalonia, it gives the following examples of specifying asset paths in AXAML:

    <ItemGroup>
        <AvaloniaResource Include="Assets\**"/>
    </ItemGroup>

...

Assets can be referenced in XAML by specifying their **relative path**:

    <Image Source="icon.png"/><Image Source="images/icon.png"/><Image Source="../icon.png"/>

Or their **rooted path**:

    <Image Source="/Assets/icon.png"/>

https://docs.avaloniaui.net/docs/getting-started/assets

Although, I am able to use this form and it works, I am left scratching my head.

For example, for "relative path", relative to what? Is this a file which must be present at runtime or is packaged in the assembly? If relative, then is it relative to the source file, the project file or the assets folder?

The second example says it is the "rooted path", but I'm pretty sure this is not my system root. I don't know what is meant by rooted?

**Comment by kekekeks** [comment]
During the build we save resource paths relative to the project file.

>If relative, then is it relative to the source file

Relative to the file the XAML is being loaded from. e. g. if you have your xaml file in `<ProjectRoot>/View/MyView.axaml` you can reference `<ProjectRoot>/Assets/icon.png` either via `../Assets/icon.png` (relative path)  or `/Assets/icon.png` (rooted path). You can reference resources from other assemblies using fully qualified uri, e. g. `avares://Avalonia.Themes.Default/DefaultTheme.xaml`
